CLAUDE.md

```diff
@@ -0,0 +1,149 @@
+| G-2 | Refactor large modules without human guidance. For changes >300 LOC or >3 files, **ask for confirmation**.                                         |
+
+---
+
+## 2. File imports
+
+NOTE: Be aware that whatever you add to this list is automatically loaded into context (due to `@` annotation). It's helpful
+   to provide project context, but only within reasonable limits.
+
+| File Import                      | Description                          |
+|----------------------------------|--------------------------------------|
+| @Makefile                        | Build, test, and formatting commands |
+| @mise.toml                       | Dependency management                |
+| @README.md                       | Main project README file             |
+| @docs/spec/src/**/*.md           | All project spec markdown files      |
+| @.claude/commands/*              | Project claude commands              |
+
+If there are imports that are relevant only to a particular part of the project, then they should be added to a CLAUDE.md
+   file *in the relevant subdirectory*. Then the imports will only be processed when files within that directory are read.
+
+---
+
+## 3. Coding standards
+
+1. **Line Length**: 120 characters. This should also be observed for Markdown files, where manual newlines should be inserted.
+   - The line length limit may be ignored in cases where a line break hinders readability, e.g. URLs.
+2. **Imports**: Use `goimports` for sorted imports. Group standard library, third-party, and local imports.
+3. **Testing**: `*_test.go` files in same package; use `testify` for assertions.
+
+### 3.1. Error handling
+
+1. Return errors explicitly; don't panic except for unrecoverable errors
+   - Some exceptions can be made for test code, where returning an error adds more complexity than benefit.
+2. Use error wrapping with `fmt.Errorf("context: %w", err)` for additional context
+   - Ensure that `%w` is used for error wrapping, *not* `%v`
+
+### 3.2. Code Documentation
+
+1. Write docs for all exported functions/types in production code
+2. Write docs for unexported functions/types if they contain non-trivial logic. A good rule of thumb: if you can't understand everything
+   there is to know about a function/type by its *name*, you should write a doc.
+3. Function/type docs should NOT simply be a rephrasing of the function/type name.
+   - E.g. the doc for `computeData` should NOT be "Computes the data".
+4. Function docs should consider the following helpful information, if relevant:
+   - What are the inputs?
+   - Are there any restrictions on what the input values are permitted to be?
+   - What is returned in the standard case?
+   - What is returned in the error case(s)?
+   - What side effects does calling the function have?
+   - Are there any performance implications that users should be aware of?
```

Comment:
Would it make sense to add a bullet point - what performance optimisations can be applied?

Reply:
a1ec752e34e6091d78c1562e0255cb495a1a1954